pyteal/compiler/compiler.py

```diff
@@ -145,17 +145,24 @@ def compileSubroutine(
     subroutineBlocks[currentSubroutine] = start
 
     referencedSubroutines: Set[SubroutineDefinition] = set()
+    subroutineCallbacks: Dict[SubroutineDefinition, SubroutineCall] = {}
     for stmt in teal:
         for subroutine in stmt.getSubroutines():
             referencedSubroutines.add(subroutine)
+            assert isinstance(
```

Comment:
This new assertion would break any previous code that has a misbehaving `getSubroutines()` method